File: Difficulty: Easy/Shop in Candy Store/shop-in-candy-store.cpp
class Solution {
  public:
    vector<int> minMaxCandy(vector<int>& prices, int k) {
        // Code here
        int N=prices.size();
        sort(prices.begin(),prices.end());
        int i=0;
        int j=N-1;
        int ans=0;
        while(i<=j)
        {
            ans+=prices[i];
            i++;
            j-=k;
        }
        vector<int>res;
        res.push_back(ans);
        i=N-1;j=0;
        ans=0;
        while(i>=j)
        {
            ans+=prices[i];
            i--;
            j+=k;
        }
        res.push_back(ans);
        return res;
    }
};

File: Difficulty: Easy/Shop in Candy Store/shop-in-candy-store.js
/**
 * @param {number[]} prices
 * @param {number} k
 * @returns {number[]}
 */
class Solution {
    minMaxCandy(prices, k) {
        let i=0;
        let j=prices.length-1;
        let ans=[];
        prices.sort((a,b)=>a-b);
        let minValue=0;
        while(i<=j)
        {
            minValue+=prices[i];
            i++;
            j-=k;
        }
        ans.push(minValue);
        i=prices.length-1;
        j=0;
        let maxValue=0;
        while(i>=j)
        {
            maxValue+=prices[i];
            i--;
            j+=k;
        }
        ans.push(maxValue);
        return ans;
    }
}
